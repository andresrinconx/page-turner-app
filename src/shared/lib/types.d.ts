import { COLORS } from "./constants";
import type { ViewStyle, TextStyle } from "react-native";

declare global {
  type FlexProps = {
    flex?: number;
    flexDirection?: ViewStyle["flexDirection"];
    justifyContent?: ViewStyle["justifyContent"];
    alignItems?: ViewStyle["alignItems"];
    flexWrap?: ViewStyle["flexWrap"];
    gap?: number;
  };

  type SpacingProps = {
    p?: number;
    px?: number;
    py?: number;
    pt?: number;
    pb?: number;
    pl?: number;
    pr?: number;
    m?: number;
    mx?: number;
    my?: number;
    mt?: number;
    mb?: number;
    ml?: number;
    mr?: number;
  };

  type PositionProps = {
    position?: ViewStyle["position"];
    top?: number;
    bottom?: number;
    left?: number;
    right?: number;
  };

  type SizingProps = {
    w?: number | string;
    h?: number | string;
    minW?: number | string;
    minH?: number | string;
    maxW?: number | string;
    maxH?: number | string;
  };

  type ColorProps = {
    bg?: keyof typeof COLORS;
    color?: keyof typeof COLORS;
  };

  type TypographyProps = {
    fontSize?: number;
    lineHeight?: number;
    fontWeight?: TextStyle["fontWeight"];
    textAlign?: TextStyle["textAlign"];
    fontFamily?: string;
    letterSpacing?: number;
    textDecorationLine?: TextStyle["textDecorationLine"];
    textTransform?: TextStyle["textTransform"];
  };

  type BorderProps = {
    borderWidth?: number;
    borderTopWidth?: number;
    borderBottomWidth?: number;
    borderLeftWidth?: number;
    borderRightWidth?: number;
    borderRadius?: number;
    borderTopLeftRadius?: number;
    borderTopRightRadius?: number;
    borderBottomLeftRadius?: number;
    borderBottomRightRadius?: number;
    borderStyle?: ViewStyle["borderStyle"];
    borderColor?: keyof typeof COLORS;
  };

  type CommonStyleProps = FlexProps &
    SpacingProps &
    SizingProps &
    PositionProps &
    ColorProps &
    TypographyProps &
    BorderProps;
}

export {};
