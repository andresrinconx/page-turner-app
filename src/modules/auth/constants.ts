/**
 * constants.ts
 * Constants for the auth module
 *
 * Created by Andres Rincon on 21/4/25.
 */
