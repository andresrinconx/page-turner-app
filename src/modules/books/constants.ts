/**
 * constants.ts
 * Constants for the books module
 *
 * Created by Andres Rincon on 21/4/25.
 */
