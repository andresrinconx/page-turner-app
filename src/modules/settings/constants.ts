/**
 * constants.ts
 * Settings module constants
 *
 * Created by Andres Rincon on 28/4/25.
 */

import { WeekdayItem } from "./types";

export const DEFAULT_WEEKDAYS: WeekdayItem[] = [
  { id: 1, label: "M", selected: true },
  { id: 2, label: "T", selected: true },
  { id: 3, label: "W", selected: true },
  { id: 4, label: "T", selected: true },
  { id: 5, label: "F", selected: true },
  { id: 6, label: "S", selected: true },
  { id: 7, label: "S", selected: true },
];

export const STORAGE_KEYS = {
  WEEKDAYS: "notifications_weekdays",
  START_DATE: "notifications_start_date",
  REMINDER_TIME: "notifications_reminder_time",
};

export const NOTIFICATION_DEFAULTS = {
  TITLE: "Page Turner",
  BODY: "Time to read your book today!",
  HOUR: 8,
  MINUTE: 0,
};
