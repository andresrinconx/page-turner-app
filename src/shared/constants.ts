/**
 * constants.ts
 * Constants for the shared components
 *
 * Created by Andres Rincon on 21/4/25.
 */

export const COLORS = {
  primary: "#0BB589",
  primaryLight: "#E4F7ED",
  secondary: "#EF844C",
  background: "#F7FBFC",
  error: "#B12B2C",
  success: "#219653",
  info: "#00CFE8",
  overlay: "#696969",
  text: "#5C605E",
  secondaryText: "#82868B",
  disabled: "#B9BBC3",
  progress: "#E0E4E5",
  inputText: "#EFEFEF",
  card: "#F7F9FD",
  white: "#FFFFFF",
  black: "#000000",
};

export const SIZES = {
  borderRadius: 16,
};
