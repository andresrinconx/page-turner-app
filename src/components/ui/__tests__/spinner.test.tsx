import { render } from "@testing-library/react-native";
import Spinner from "../spinner";
import { COLORS } from "../../../lib/constants";

describe("Spinner", () => {
  it("renders correctly", () => {
    const { UNSAFE_getByProps } = render(<Spinner />);
    const spinner = UNSAFE_getByProps({ size: "small" });
    expect(spinner).toBeTruthy();
  });

  it("has correct props", () => {
    const { UNSAFE_getByProps } = render(<Spinner />);
    const spinner = UNSAFE_getByProps({ size: "small" });
    expect(spinner.props.size).toBe("small");
    expect(spinner.props.color).toBe(COLORS.primary);
  });
});
