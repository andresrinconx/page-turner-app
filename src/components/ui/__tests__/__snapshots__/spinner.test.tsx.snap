// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner matches snapshot 1`] = `
<ActivityIndicator
  color="#0BB589"
  size="small"
/>
`;
